plan/physical_plan_builder.go

```diff
@@ -1048,6 +1048,58 @@ func (p *Distinct) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlanIn
 	return info, nil
 }
 
+// convert2PhysicalPlan implements the LogicalPlan convert2PhysicalPlan interface.
+func (p *Analyze) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlanInfo, error) {
+	info, err := p.getPlanInfo(prop)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if info != nil {
+		return info, nil
+	}
+	var childInfos []*physicalPlanInfo
+	for _, idx := range p.IdxOffsets {
+		var columns []*model.ColumnInfo
+		tblInfo := p.Table.TableInfo
+		for _, idxCol := range tblInfo.Indices[idx].Columns {
+			for _, col := range tblInfo.Columns {
+				if col.Name.L == idxCol.Name.L {
+					columns = append(columns, col)
+					break
+				}
+			}
+		}
+		is := &PhysicalIndexScan{
+			Index:               tblInfo.Indices[idx],
+			Table:               tblInfo,
+			Columns:             columns,
+			TableAsName:         &p.Table.Name,
+			OutOfOrder:          false,
+			DBName:              &p.Table.DBInfo.Name,
+			physicalTableSource: physicalTableSource{client: p.ctx.GetClient()},
+			DoubleRead:          false,
+		}
+		is.tp = Aly
+		is.allocator = p.allocator
+		is.initIDAndContext(p.ctx)
+		is.SetSchema(expression.TableInfo2Schema(tblInfo))
+		is.readOnly = true
+		rb := rangeBuilder{sc: p.ctx.GetSessionVars().StmtCtx}
+		is.Ranges = rb.buildIndexRanges(fullRange, types.NewFieldType(mysql.TypeNull))
+		childInfos = append(childInfos, is.matchProperty(prop, &physicalPlanInfo{count: 0}))
+	}
+	for _, child := range p.GetChildren() {
```

Comment:
But it may have 2 table scan, for example, when the pk is handle, so loop is necessary.
Also, construct the PhysicalPlanInfo does no harm and makes the code easy to understand.